src/libstd/sys/unix/process.rs

```diff
@@ -288,6 +288,31 @@ impl Process {
             unsafe { libc::_exit(1) }
         }
 
+        // Make sure that the source descriptors are not an stdio descriptor,
+        // otherwise the order which we set the child's descriptors may blow
+        // away a descriptor which we are hoping to save. For example,
+        // suppose we want the child's stderr to be the parent's stdout, and
+        // the child's stdout to be the parent's stderr. No matter which we
+        // dup first, the second will get overwritten prematurely.
+        let maybe_migrate = |src: Stdio, output: &mut AnonPipe| {
+            match src {
+                Stdio::Raw(fd@libc::STDIN_FILENO) |
+                Stdio::Raw(fd@libc::STDOUT_FILENO) |
+                Stdio::Raw(fd@libc::STDERR_FILENO) => match cvt_r(|| libc::dup(fd)) {
+                    Err(_) => fail(output),
+                    Ok(fd) => {
+                        let fd = RawStdio::new(fd);
```

Comment:
Could this use `FileDesc` instead of `RawStdio` here? (as `RawStdio` is just a type alias)